### PR TITLE
Improve parameter type inference error messaging

### PR DESCRIPTION
**Pull request by Sporarum**
We fixed the issue, as well as hopefully improved the existing error

Done with @sankalpgambhir

### CODE REVIEW

File: tests/neg/i17183.check [hunk 1]
```diff
@@ -1,14 +1,18 @@
 -- [E081] Type Error: tests/neg/i17183.scala:11:24 ---------------------------------------------------------------------
 11 |def test = Context(f = (_, _) => ???) // error // error
    |                        ^
-   |                        Missing parameter type
+   |                        Could not infer type for parameter _$1 of anonymous function
```
Comment (nicolasstucki) [inline tests/neg/i17183.check:4]: I wonder if it would be possible to report
```
 Could not infer type for the 1st parameter of an anonymous function
```

Comment (Sporarum) [reply tests/neg/i17183.check:4]: I think it would be possible, but I'm not sure it's desirable, I think here we need to show some internals

Comment (nicolasstucki) [reply tests/neg/i17183.check:4]: The internal names of `_` and other unamed variables are only useful to us. Never to the user as they cannot use them in the source.

Comment (Sporarum) [reply tests/neg/i17183.check:4]: I agree, but we need to show the desugarring anyways, as the issue might be that the user missunderstood how `_`works, so we need names for these identifiers anyways

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**